src/index.ts

```diff
@@ -1281,7 +1479,9 @@ export abstract class PinejsClientCore<
 		return new Poll(requestFn, pollInterval);
 	}
 
-	public put(params: Params): Promise<void> {
+	public put<TResource extends StringKeyOf<Model>>(
+		params: { resource: TResource } & Params<Model[TResource]>,
```

Comment:
I am not sure this mandatory resource is valid everytime as a client can pass the URL directly and bypass resource entirely. I think resource/url would be alternate resources e.g ({resource: TResource} | {url: string}) & Params<Model[TResource]>

Same is valid for all the others methods expecting resource directly